scripts/deploy_local_kind_registry.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+set -o errexit
+
+# Some of this copy-pasted from https://kind.sigs.k8s.io/docs/user/local-registry/
+
+# 1. Create registry container unless it already exists
+reg_name='local-mr-registry'
+reg_port='5001'
+if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --network bridge --name "${reg_name}" \
+    registry:2
+fi
```

Comment:
interesting approach, thank you for sharing; which pro/cons if using again distribution registry but as a bespoke deployment in olot?

Reply:
btw, not sure if the e2e failure is related: 

```
----------------------------- Captured stderr call -----------------------------
time="2025-02-14T14:55:54Z" level=fatal msg="copying image 1/2 from manifest list: trying to reuse blob sha256:1933e30a3373776d5c7155591a6dacbc205cf6a2665b6dced682c6d2ea7b000f at destination: pinging container registry localhost:5001: Get \"http://localhost:5001/v2/\": dial tcp [::1]:5001: connect: connection refused"
```

Reply:
Yes that failure could be related, I'll look into it. 

>which pro/cons if using again distribution registry but as a bespoke deployment in olot?

Could you elaborate on this? I'm not quite following what you mean

Reply:
> Could you elaborate on this? I'm not quite following what you mean

in olot/omlmd, we use a CNCF Distribution Registry deployed as a K8s workload.

- https://github.com/containers/olot/blob/main/e2e/deploy_distribution_registry.sh
- https://github.com/containers/omlmd/blob/a1b2030ed39355dfa6ff93cdc118c47f1e210b70/.github/workflows/e2e.yaml#L10-L33

here it seems to me you're running it "barebone" outside the cluster (so not sure which defaults are applied).

Reply:
Understood! I decided to run it outside of the kind cluster just for simplicity, but I can take a look at how its done in olot and match it here. That will provide some consistency